prow/sidecar/run_test.go

```diff
@@ -252,40 +398,46 @@ func name(idx int) string {
 
 func TestLogReader(t *testing.T) {
 	cases := []struct {
-		name     string
-		pieces   []string
-		expected []string
+		name        string
+		processLogs map[string]string
+		expected    map[string]string
 	}{
 		{
-			name:     "basically works",
-			pieces:   []string{"hello world"},
-			expected: []string{"hello world"},
+			name: "basically works",
+			processLogs: map[string]string{
+				"process-log.txt": "hello world",
+			},
+			expected: map[string]string{
+				"build-log.txt": "hello world",
+			},
 		},
 		{
-			name:   "multiple logging works",
-			pieces: []string{"first", "second"},
-			expected: []string{
-				start(name(0)),
-				"first",
-				start(name(1)),
-				"second",
+			name: "multiple logs works",
+			processLogs: map[string]string{
+				"test1-log.txt": "hello",
+				"test2-log.txt": "world",
+			},
+			expected: map[string]string{
+				"test1-build-log.txt": "hello",
+				"test2-build-log.txt": "world",
 			},
 		},
 		{
-			name:   "note when a part has aproblem",
-			pieces: []string{"first", "missing", "third"},
-			expected: []string{
-				start(name(0)),
-				"first",
-				start(name(1)),
-				"Failed to open log-1.txt: whatever\n",
-				start(name(2)),
-				"third",
+			name: "note when a part has a problem",
+			processLogs: map[string]string{
+				"test1-log.txt": "hello",
+				"test2-log.txt": "missing",
+				"test3-log.txt": "world",
+			},
+			expected: map[string]string{
+				"test1-build-log.txt": "hello",
+				"test2-build-log.txt": "Failed to open test2-log.txt: whatever\n",
+				"test3-build-log.txt": "world",
 			},
 		},
 	}
 
-	re := regexp.MustCompile(`(?m)(Failed to open) .*log-\d.txt: .*$`)
+	re := regexp.MustCompile(`(?m)(Failed to open) .*log\.txt: .*$`)
```

Comment:
`regexp.MustCompile` should only be used to declare package variables since it will result in a runtime panic and it is better if that is discovered at init time rather than when we try to use the regexp. I know this isn't an addition you made, but please move this outside the function in order to make it a package variable.